### PR TITLE
Sass 1528: Create childcare employment benefits radio button EOY page

### PR DESCRIPTION
**Pull request by joeyymoore**
### Description

- Create childcare employment benefits radio button EOY page
- Link the page to the CYA childcare change link

https://jira.tools.tax.service.gov.uk/browse/SASS-1528

##### Before Reviewing
- [ ]  Have you pulled the branch down?
- [ ]  Have you assigned yourself to the PR?
- [ ]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [ ]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

**Review by HarryGull — CHANGES_REQUESTED** [review]

File: app/controllers/benefits/medicalChildcareEducation/ChildcareBenefitsController.scala [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                employmentId,
+                updatedCyaModel,
+                taxYear,
+                data.isPriorSubmission,
+                data.hasPriorBenefits
+              )(errorHandler.internalServerError()) {
+                if (data.isPriorSubmission) {
+                  Redirect(CheckYourBenefitsController.show(taxYear, employmentId))
+                } else {
+                  Redirect(CheckYourBenefitsController.show(taxYear, employmentId))
```
[the remark this answers is not in the document]
Comment (joeyymoore) [reply app/controllers/benefits/medicalChildcareEducation/ChildcareBenefitsController.scala:99]: good idea

File: it/controllers/benefits/medicalChildcareEducation/ChildcareBenefitsControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+    UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN, Some(ExpectedIndividualEN)),
+    UserScenario(isWelsh = false, isAgent = true, CommonExpectedEN, Some(ExpectedAgentEN)),
+    UserScenario(isWelsh = true, isAgent = false, CommonExpectedCY, Some(ExpectedIndividualCY)),
+    UserScenario(isWelsh = true, isAgent = true, CommonExpectedCY, Some(ExpectedAgentCY))
+  )
+
+  ".show" should {
+    userScenarios.foreach { user =>
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+        "render the 'Did you get any benefits for other services' page with the correct content without pre-filled form" which {
```
[the remark this answers is not in the document]
Comment (joeyymoore) [reply it/controllers/benefits/medicalChildcareEducation/ChildcareBenefitsControllerISpec.scala:144]: wow that was lazy of me, good spot

File: it/controllers/benefits/medicalChildcareEducation/ChildcareBenefitsControllerISpec.scala [hunk 3]
```diff
@@ -0,0 +1,493 @@
...
+          linkCheck(expectedWeOnlyLink, onlyNeedLinkSelector, exemptLink)
+          textOnPageCheck(user.specificExpectedResults.get.expectedCheckWith, paragraphSelector(4))
+          radioButtonCheck(yesText, 1, Some(false))
+          radioButtonCheck(noText, 2, Some(false))
+          buttonCheck(expectedButtonText, continueButtonSelector)
+          formPostLinkCheck(continueLink, continueButtonFormSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "render the 'Did you get any benefits for other services' page with the correct content with yes value pre-filled" which {
```
[the remark this answers is not in the document]
Comment (joeyymoore) [reply it/controllers/benefits/medicalChildcareEducation/ChildcareBenefitsControllerISpec.scala:176]: wow that was lazy of me, good spot

File: it/controllers/benefits/medicalChildcareEducation/ChildcareBenefitsControllerISpec.scala [hunk 4]
```diff
@@ -0,0 +1,493 @@
...
+          linkCheck(expectedWeOnlyLink, onlyNeedLinkSelector, exemptLink)
+          textOnPageCheck(user.specificExpectedResults.get.expectedCheckWith, paragraphSelector(4))
+          radioButtonCheck(yesText, 1, Some(true))
+          radioButtonCheck(noText, 2, Some(false))
+          buttonCheck(expectedButtonText, continueButtonSelector)
+          formPostLinkCheck(continueLink, continueButtonFormSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "render the 'Did you get any benefits for other services' page with the correct content with no value pre-filled" which {
```
[the remark this answers is not in the document]
Comment (joeyymoore) [reply it/controllers/benefits/medicalChildcareEducation/ChildcareBenefitsControllerISpec.scala:208]: wow that was lazy of me, good spot

File: it/controllers/benefits/medicalChildcareEducation/ChildcareBenefitsControllerISpec.scala [hunk 5]
```diff
@@ -0,0 +1,493 @@
...
+        }
+      }
+
+    }
+  }
+
+  ".submit" should {
+    userScenarios.foreach { user =>
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+        "should render 'Did you get any benefits for other services' page with empty error text when there no input" which {
```
[the remark this answers is not in the document]
Comment (joeyymoore) [reply it/controllers/benefits/medicalChildcareEducation/ChildcareBenefitsControllerISpec.scala:314]: wow that was lazy of me, good spot

File: it/controllers/benefits/medicalChildcareEducation/ChildcareBenefitsControllerISpec.scala [hunk 6]
```diff
@@ -0,0 +1,493 @@
...
+        insertCyaData(employmentUserData(hasPriorBenefits = false, cyaModel("employerName", hmrc = true,
+          benefits = Some(benefits(fullMedicalChildcareEducationModel.copy(nurseryPlacesQuestion = Some(false)))))), userRequest)
+
+        urlPost(pageUrl(taxYearEOY), body = form, follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+      }
+
+      "has an SEE_OTHER status" in {
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe
+          Some(s"/income-through-software/return/employment-income/$taxYearEOY/check-employment-benefits?employmentId=$employmentId")
```
[the remark this answers is not in the document]
Comment (joeyymoore) [reply it/controllers/benefits/medicalChildcareEducation/ChildcareBenefitsControllerISpec.scala:415]: good idea

**Review by HarryGull — APPROVED** [review]

**Pull request merged**